cpp/arcticdb/storage/s3/detail-inl.hpp

```diff
@@ -125,6 +125,26 @@ namespace s3 {
                    || err == Aws::S3::S3Errors::RESOURCE_NOT_FOUND;
         }
 
+        inline void raise_if_unexpected_error(const Aws::S3::S3Error& err){
+            if (!is_expected_error_type(err.GetErrorType())) {
+                // We log a more detailed error explanation in case of NETWORK_CONNECTION errors to remedy #880.
+                if (err.GetErrorType() == Aws::S3::S3Errors::NETWORK_CONNECTION){
+                    log::storage().error("Got an unexpected network error: {}. "
+                                         "This could be due to a connectivity issue or exhausted open sockets. "
+                                         "In case socket file descriptors are exhausted, consider increasing `ulimit -n`.",
```

Comment:
Ah I misremembered the message. We should, like @alexowens90 said, suggest creating fewer `Arctic` objects in the first instance (the system works best with only one per connection URI live in memory at a time).